yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,561 @@
+        return {
+            'id': video_id,
+            'title': str_or_none(traverse_obj(downloaded_json_meta_data, ('content', 'contentTitle'))),
+            'webpage_url': self._RECOMMENDED_VIDEO_BASE_URL + video_id,
+            # webpage_url = url_from_user minus the final part. The final part exists solely for human consumption and is otherwise irrelevant.
+            'live_status': 'not_live',
+            'duration': float_or_none(traverse_obj(downloaded_json_meta_data, ('content', 'duration'))),
+            'thumbnail': url_or_none(traverse_obj(downloaded_json_meta_data, ('content', 'thumbnailUrl1'))),
+            'description': str_or_none(traverse_obj(downloaded_json_meta_data, ('content', 'contentDescription'))),
+            'like_count': int_or_none(downloaded_json_meta_data.get('likeCount')),
+            'dislike_count': int_or_none(downloaded_json_meta_data.get('dislikeCount')),
+            # 'comment_count': downloaded_json_meta_data.get('numComments'), # Uncomment when Rf corrects 'numComments' field.
+            'availability': availability,
+            'creator': str_or_none(traverse_obj(downloaded_json_meta_data, ('createdBy', 'name'))),
+            'channel_id': traverse_obj(downloaded_json_meta_data, ('createdBy', 'id')),
+            'channel': str_or_none(traverse_obj(downloaded_json_meta_data, ('createdBy', 'name'))),
+            'channel_url': try_get(downloaded_json_meta_data, lambda x: url_or_none(self._CHANNEL_BASE_URL + x['createdBy']['username'])),
+            'timestamp': unified_timestamp(downloaded_json_meta_data.get('creationDateTime')),
+            'tags': downloaded_json_meta_data.get('tags') or [],
+            'formats': frmts or [],
+            'subtitles': subs or {},
+            '__post_extractor': self.extract_comments(video_id=video_id)
+        }
+
+    def _get_comments(self, video_id):
+        _COMMENTS_BASE_URL = 'https://prod-api-v2.production.rokfin.com/api/v2/public/comment'
+        _COMMENTS_PER_REQUEST = 50  # 50 is the maximum Rokfin permits per request.
+
+        def dnl_comments_incrementally(base_url, video_id, comments_per_page):
+            pages_total = None
+
+            for page_n in itertools.count(0):
+                raw_comments = self._download_json(
+                    f'{base_url}?postId={video_id[5:]}&page={page_n}&size={comments_per_page}',
+                    video_id, note=f'Downloading viewer comments (page {page_n + 1}' + (f' of {pages_total}' if pages_total else '') + ')',
+                    fatal=False) or {}
+
+                raw_comments = raw_comments.get
+                comments = raw_comments('content')
+                pages_total = int_or_none(raw_comments('totalPages'))
+                is_last_page = bool_or_none(raw_comments('last'))
+                max_page_count_reached = None if pages_total is None else (page_n + 1 >= pages_total)
+
+                if comments:
+                    yield comments
```

Comment:
the nested functions is unnecessary and the actual comment yielding can be moved here

Reply:
Without this nested function, the code will get messier. Right now, everything is perfectly encapsulated (hence less error-prone), and there is a clear division of responsibilities: the nested function decides what elements will be looped over, while the main loop decides what to **do** with those elements. That's perfect from software engineering perspective.